schemas/event.yml

```diff
@@ -704,3 +704,19 @@
         Alert events, indicated by `event.kind:alert`, are a common use case for this field.
 
       example: https://mysystem.mydomain.com/alert/5271dedb-f5b0-4218-87f0-4ac4870a38fe
+
+    - name: reason
+      level: extended
+      type: keyword
+      short: The reason captured by the event
+      description: >
+        The reason captured by the event.
+
+        This describes the why of a particular action or outcome captured in the event. Where
+        `event.action` captures the action from the event, `event.reason` describes why that action
+        was taken. For example, a web proxy with an `event.action` which denied the request may also
+        populate `event.reason` with the reason why (e.g. `blocked site`).
+      example: "Terminated an unexpected process"
+      multi_fields:
+      - type: text
+        name: text
```

Comment:
Do we expect `event.reason` to be complex often enough to warrant this? Not opposed to this, just wondering out loud :-)

Reply:
I had a similar internal debate...

I can see value in having the `text` indexing available since `event.reason` could possibly be longer string (e.g. something like [this Okta event](https://github.com/elastic/beats/blob/master/x-pack/filebeat/module/okta/system/test/okta-system-test.json.log-expected.json#L196)), but the extra complexity may not be warranted the majority of the time.

As I type this out, I'm leaning towards removing the multi-field. If a user was trying to search for allowed events across sources using `event.reason.text:allow`, using the categorization fields like `event.type:"allowed" would be the better practice to encourage I think.

Reply:
Let's remove it for now. Adding `.text` later is non-breaking, so it'll be fine if it becomes necessary.

Also, I think this will be one of the fields we move to `wildcard` ultimately. I'm sure we'll get to `wildcard` before there's a need to add the `.text` interim solution.